Dockerfile

```diff
@@ -0,0 +1,63 @@
+# syntax=docker/dockerfile:1.4
+
+ARG BASE_IMAGE="python:3.11-slim-bookworm"
+
+FROM ${BASE_IMAGE}
+
+ARG PYTORCH_INSTALL_ARGS=""
+ARG EXTRA_ARGS=""
+ARG USERNAME="comfyui"
+ARG USER_UID=1000
+ARG USER_GID=${USER_UID}
+
+RUN \
+	--mount=target=/var/lib/apt/lists,type=cache,sharing=locked \
+	--mount=target=/var/cache/apt,type=cache,sharing=locked \
+	set -eux; \
+		apt-get update; \
+		apt-get install -y --no-install-recommends \
+			git \
+			git-lfs \
+ 			rsync
+
+RUN set -eux; \
+	groupadd --gid ${USER_GID} ${USERNAME}; \
+	useradd --uid ${USER_UID} --gid ${USER_GID} -m ${USERNAME}
+
+# run instructions as user
+USER ${USER_UID}:${USER_GID}
+
+WORKDIR /app
+
+ENV PIP_CACHE_DIR="/cache/pip"
+ENV VIRTUAL_ENV=/app/venv
+ENV VIRTUAL_ENV_CUSTOM=/app/custom_venv
+ENV TRANSFORMERS_CACHE="/app/.cache/transformers"
+
+# create cache directory
+RUN mkdir -p ${TRANSFORMERS_CACHE}
+
+# create virtual environment to manage packages
+RUN python -m venv ${VIRTUAL_ENV}
+
+# run python from venv
+ENV PATH="${VIRTUAL_ENV_CUSTOM}/bin:${VIRTUAL_ENV}/bin:${PATH}"
+
+RUN --mount=type=cache,target=/cache/,uid=${USER_UID},gid=${USER_GID} \
+	pip install torch torchvision torchaudio ${PYTORCH_INSTALL_ARGS}
+
+# copy requirements files first so packages can be cached separately
+COPY --chown=${USER_UID}:${USER_GID} requirements.txt .
+RUN --mount=type=cache,target=/cache/,uid=${USER_UID},gid=${USER_GID} \
+	pip install -r requirements.txt
+
+COPY --chown=${USER_UID}:${USER_GID} . .
+
+# default environment variables
+ENV COMFYUI_ADDRESS=0.0.0.0
+ENV COMFYUI_PORT=8188
```

Comment:
How likely are these to need changing? Are these ENV used for anything else beyond the final CMD in this image?

In most cases the docker container will receive connections from the network interface it's attached to, as `127.0.0.1` wouldn't be reachable outside of the container itself (_unless using host mode networking_). So the only benefit to adjust that is if there's a specific interface (visible to the container) that you want to listen on instead of all interfaces it's aware of?

Provided you're not using host mode networking (_which doesn't work well in WSL on windows, nor macOS I think?_), you'd alternatively restrict interfaces on the docker host to bind published ports to (_such as `127.0.0.1` which would be suitable for keeping it only accessible locally_).

Since you could use the approach with `CMD` to customize with `--listen` like mentioned on my review feedback for that, I don't think this ENV is needed. Same applies for `COMFYUI_PORT`, it's internal with container networking, so the user should only really care about the published port which can be whatever they choose, and that will map to the internal port which could just be `80` to better convey it's an HTTP-only port.